src/prism.c

```diff
@@ -12824,6 +12855,8 @@ parse_variable_call(pm_parser_t *parser) {
                 pm_parser_local_add_token(parser, &parser->previous);
                 return (pm_node_t *) pm_local_variable_read_node_create(parser, &parser->previous, 0);
             }
+        } else if ((parser->version != PM_OPTIONS_VERSION_CRUBY_3_3_0) && !parser->current_scope->closed && pm_token_is_it(parser->previous.start, parser->previous.end)) {
```

Comment:
Maybe the test should be `parser->version >= PM_OPTIONS_VERSION_CRUBY_3_4_0`?
Because e.g. for 3.3.1 we wouldn't want to handle `it` specially.

Reply:
For that, the enum would need to look like:

```c
typedef enum {
    PM_OPTIONS_VERSION_CRUBY_3_3_0 = 0,
    PM_OPTIONS_VERSION_CRUBY_3_4_0 = 1,
    PM_OPTIONS_VERSION_LATEST = 2
} pm_options_version_t;
```

However, even having the following enum on top of `main` (swapping 0 and 1 between 3_3_0 and LATEST) breaks a lot of stuff for some reason:

```c
typedef enum {
    PM_OPTIONS_VERSION_CRUBY_3_3_0 = 0,
    PM_OPTIONS_VERSION_LATEST = 1
} pm_options_version_t;
```

It doesn't seem like a trivial change, so I think it should be addressed in a separate PR.

Reply:
Right.
Do you have a link to a failed job?
Maybe the issue is these constants are hardcoded in FFI too: https://github.com/ruby/prism/blob/829ac0ed3f449313584aae35db98fd7614eb9d63/lib/prism/ffi.rb#L318

It does seem better to avoid changing the values of the enum to avoid tricky missing recompilation issues, I was thinking `PM_OPTIONS_VERSION_CRUBY_3_3_0 = 330` would work nicely, and `PM_OPTIONS_VERSION_LATEST = 999`.